### PR TITLE
New module system

### LINKED ISSUE #59: Modules should be separable
While writing the test for `PrognosticData` I came across the need to include all the modules from the physics subdirectory when just trying to write code only involving core classes.

Some way needs to be available to only use the core (or only the physics) modules.

Currently this is fixed by using alternative hand-written `ipp` files when compiling the tests.

### PR DESCRIPTION
**Pull request by timspainNERSC**
Python free and more self-contained.

Will close #59.

### CODE REVIEW

**Comment by timspainNERSC** [comment]
Well, python free in the important sense. I am writing a script to auto generate the boiler-plate heavy module class files.

File: core/src/modules/README.md [hunk 1]
```diff
@@ -1,75 +0,0 @@
-# ModuleLoader
```
Comment (einola) [inline core/src/modules/README.md:1]: Don't we still want a README here?

**Review by einola — CHANGES_REQUESTED** [review]
Looks good to me. But I miss that README file. There should still be one to tell developers how to develop a new module.

File: core/src/modules/README.md (hunk 1 above)
Comment (timspainNERSC) [reply core/src/modules/README.md:1]: Almost certainly.

**Review by einola — APPROVED** [review]

**Pull request merged**